# Databricks Contact Center Analysis
### 10/01/2022
### Fasulo Matteo, Paris Simone Flavio, Sivoccia Matteo

In [0]:
# Ci avvaliamo dell'uso di Pandas per l'alterazione dei dataset del contact center
import pandas as pd

## Creazione dei DataFrame e Temporary Table (Punti 4 e 5)

### DataFrame Casi

In [0]:
# File location and type
file_location = "/FileStore/LUMSA/Casi.csv"
file_type = "csv"

# CSV options
infer_schema = "false"
first_row_is_header = "true"
delimiter = ";"

# The applied options are for CSV files. For other file types, these will be ignored.
casi = spark.read.format(file_type) \
  .option("inferSchema", infer_schema) \
  .option("header", first_row_is_header) \
  .option("sep", delimiter) \
  .load(file_location)

display(casi)

temp_table_name = "Casi_csv"

casi.createOrReplaceTempView(temp_table_name)

# Numero di osservazioni
print(f'Il DF casi contiene: {casi.count()} occorrenze')

# Poiché sia KEY_DATA che DataArrivo sono data non in formato ISO, provvedo a renderle tali per semplificare qualunque futura operazione inerente alle date
# Prima trasformo la struttura a datafram di Pandas attraverso toPandas()
CASI = casi.toPandas()
CASI['KEY_DATA'] = pd.to_datetime(CASI['KEY_DATA'],format='%Y|%m|%d')
CASI['DataArrivo'] = pd.to_datetime(CASI['DataArrivo'],format='%d/%m/%Y')
CASI

Il DF casi contiene: 149997 occorrenze


customerid 
 LivelloEscalation 
 KEY_DATA 
 OraArrivo 
 NumeroTicket 
 DataArrivo 
 OrigineCaso 
 ClassificazioneTicket 
 
 
 
 
 0 
 00006912-4290-ea11-a811-000d3a23cc7b 
 Livello 2 
 2020-05-07 
 14 
 TCK-52759-W1C0 
 2020-05-07 
 3 
 Informazioni 
 
 
 1 
 0000c765-d180-ea11-a811-000d3a21188e 
 Livello 1 
 2020-04-20 
 10 
 TCK-45038-K4C6 
 2020-04-20 
 3 
 Informazioni 
 
 
 2 
 00012cd8-3ea7-eb11-b1ac-000d3ac251b6 
 Livello 2 
 2021-04-27 
 14 
 TCK-192329-L0X9 
 2021-04-27 
 3 
 Informazioni 
 
 
 3 
 00021b17-7438-eb11-a813-000d3a3a1391 
 Livello 1 
 2020-12-07 
 11 
 TCK-130028-X9N6 
 2020-12-07 
 1 
 Informazioni 
 
 
 4 
 000285f4-9581-eb11-a812-000d3abd3d02 
 Livello 3 
 2021-03-11 
 11 
 TCK-169373-C9N9 
 2021-03-11 
 3 
 Informazioni 
 
 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 
 
 149992 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 Livello 3 
 2021-02-24 
 13 
 TCK-162029-G8M5 
 2021-02-24 
 3 
 Informazioni 
 
 
 149993 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 Livello 3 
 2021-03-17 
 12 
 TCK-172017-X1L7 
 2021-03-17 
 3 
 Informazioni 
 
 
 149994 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 Livello 3 
 2021-04-23 
 16 
 TCK-190698-G3J7 
 2021-04-23 
 3 
 Informazioni 
 
 
 149995 
 8ff976a2-4be4-e911-a813-000d3a24e3ab 
 Livello 1 
 2020-05-25 
 13 
 TCK-61020-V9N8 
 2020-05-25 
 1 
 Informazioni 
 
 
 149996 
 8ff976a2-4be4-e911-a813-000d3a24e3ab 
 Livello 1 
 2020-06-09 
 17 
 TCK-65209-R6X4 
 2020-06-09 
 1 
 Informazioni 
 
 
 
 149997 rows × 8 columns

customerid LivelloEscalation KEY_DATA OraArrivo NumeroTicket DataArrivo OrigineCaso ClassificazioneTicket 00006912-4290-ea11-a811-000d3a23cc7b Livello 2 2020|05|07 14 TCK-52759-W1C0 07/05/2020 3 Informazioni 0000c765-d180-ea11-a811-000d3a21188e Livello 1 2020|04|20 10 TCK-45038-K4C6 20/04/2020 3 Informazioni 00012cd8-3ea7-eb11-b1ac-000d3ac251b6 Livello 2 2021|04|27 14 TCK-192329-L0X9 27/04/2021 3 Informazioni 00021b17-7438-eb11-a813-000d3a3a1391 Livello 1 2020|12|07 11 TCK-130028-X9N6 07/12/2020 1 Informazioni 000285f4-9581-eb11-a812-000d3abd3d02 Livello 3 2021|03|11 11 TCK-169373-C9N9 11/03/2021 3 Informazioni 00059e95-8ef6-eb11-94ef-00224884aa9c Livello 1 2021|08|06 13 TCK-235836-P9Y2 06/08/2021 3 Informazioni 00074734-abdc-4445-85ed-d6ae3e0d9ef7 Livello 2 2021|11|09 12 TCK-277429-D3R2 09/11/2021 3 Reclamo 00078c34-3886-ea11-a811-000d3a21188e Livello 1 2020|04|24 16 TCK-46359-P3V1 24/04/2020 1 Informazioni 00078c34-3886-ea11-a811-000d3a21188e Livello 1 2020|04|24 17 TCK-46376-T8V7 24/04/2020 1 Informazioni 00078c34-3886-ea11-a811-000d3a21188e Livello 1 2020|05|08 12 TCK-53854-T3F0 08/05/2020 1 Informazioni 00078c34-3886-ea11-a811-000d3a21188e Livello 1 2020|05|08 14 TCK-53999-T9Y1 08/05/2020 1 Informazioni 00078ce5-f23f-ec11-8c62-6045bd8d9ef6 Livello 2 2021|11|07 19 TCK-276239-J4L6 07/11/2021 3 Reclamo 00093f79-1169-ea11-a811-000d3a23cc7b Livello 1 2020|03|18 15 TCK-38187-C4T2 18/03/2020 3 Informazioni 00093f79-1169-ea11-a811-000d3a23cc7b Livello 1 2020|03|31 14 TCK-41301-Q7X5 31/03/2020 3 Informazioni 00093f79-1169-ea11-a811-000d3a23cc7b Livello 1 2020|04|07 14 TCK-42624-X1B8 07/04/2020 3 Informazioni 00093f79-1169-ea11-a811-000d3a23cc7b Livello 1 2020|04|14 20 TCK-44022-X1F0 14/04/2020 3 Reclamo 0009cbf5-4823-eb11-a813-000d3a23cc7b Livello 1 2020|11|10 12 TCK-116521-T8Z7 10/11/2020 3 Informazioni 0009cbf5-4823-eb11-a813-000d3a23cc7b Livello 1 2020|11|23 12 TCK-123232-V5C4 23/11/2020 1 Informazioni 0009cbf5-4823-eb11-a813-000d3a23cc7b Livello 1 2020|11|23 18 TCK-123790-T8X3 23/11/2020 3 Informazioni 0009e973-7920-eb11-a813-000d3a21188e Livello 1 2021|02|10 13 TCK-154786-S5N0 10/02/2021 1 Informazioni 0009e973-7920-eb11-a813-000d3a21188e Livello 1 2021|03|03 13 TCK-165432-H9X1 03/03/2021 1 Informazioni 000c8180-8240-eb11-a813-000d3a21188e Livello 1 2020|12|18 14 TCK-134128-T2P8 18/12/2020 1 Informazioni 000c8180-8240-eb11-a813-000d3a21188e Livello 2 2020|12|17 20 TCK-133853-F3L1 17/12/2020 3 Informazioni 000c8180-8240-eb11-a813-000d3a21188e Livello 2 2020|12|18 13 TCK-134120-H6Z4 18/12/2020 3 Informazioni 000c990c-b757-ea11-a811-000d3a24f1d2 Livello 1 2020|03|02 15 TCK-35573-G7Q7 02/03/2020 1 Informazioni 000c990c-b757-ea11-a811-000d3a24f1d2 Livello 1 2020|03|02 16 TCK-35584-Z6M0 02/03/2020 1 Informazioni 000c990c-b757-ea11-a811-000d3a24f1d2 Livello 3 2020|03|02 16 TCK-35586-C1R4 02/03/2020 3 Informazioni 000d020f-f2f5-ea11-a815-000d3a23cc7b Livello 1 2020|09|13 21 TCK-93237-T4K5 13/09/2020 3 Informazioni 000dc36f-e04f-eb11-a812-000d3a21188e Livello 1 2021|01|07 8 TCK-137742-W5Q3 07/01/2021 3 Informazioni 000dd486-476c-eb11-a812-000d3a391feb Livello 1 2021|02|11 10 TCK-155222-R2N3 11/02/2021 1 Orientamento 000dd486-476c-eb11-a812-000d3a391feb Livello 1 2021|02|11 16 TCK-155725-F6M9 11/02/2021 1 Informazioni 000dd486-476c-eb11-a812-000d3a391feb Livello 1 2021|04|27 14 TCK-192295-S3Q9 27/04/2021 1 Informazioni 000df6a2-6e9c-eb11-b1ac-000d3a2d98e3 Livello 3 2021|08|10 17 TCK-236756-J1H2 10/08/2021 3 Informazioni 000ec335-952a-eb11-a813-000d3a23cc7b Livello 1 2021|02|23 17 TCK-161592-F2J8 23/02/2021 1 Informazioni 000ec335-952a-eb11-a813-000d3a23cc7b Livello 2 2021|03|03 23 TCK-165742-J1D7 03/03/2021 3 Informazioni 00117bf2-ef19-ec11-b6e6-6045bd8de0b6 Livello 2 2021|09|20 11 TCK-250665-F7R1 20/09/2021 3 Informazioni 0011a75c-f32b-4323-8290-2990ef78682b Livello 2 2021|11|08 11 TCK-276475-V8W1 08/11/2021 3 Informazioni 0011a75c-f32b-4323-8290-2990ef78682b Livello 2 2021|11|11 17 TCK-279268-D2L5 11/11/2021 3 Informazioni 0011a75c-f

### DataFrame Calendario

In [0]:
# File location and type
file_location = "/FileStore/LUMSA/Calendario.csv"
file_type = "csv"

# CSV options
infer_schema = "false"
first_row_is_header = "true"
delimiter = ";"

# The applied options are for CSV files. For other file types, these will be ignored.
calendario = spark.read.format(file_type) \
  .option("inferSchema", infer_schema) \
  .option("header", first_row_is_header) \
  .option("sep", delimiter) \
  .load(file_location)

display(calendario)

temp_table_name = "Calendario_csv"

calendario.createOrReplaceTempView(temp_table_name)

# Numero di osservazioni
print(f'Il DF calendario contiene: {calendario.count()} occorrenze')

# Anche qui è presente un campo datetime che non è ISO-format...
Calendario_DAY = calendario.toPandas()
Calendario_DAY['KEY_DATA'] = pd.to_datetime(Calendario_DAY['KEY_DATA'],format='%Y|%m|%d')
Calendario_DAY

Il DF calendario contiene: 856 occorrenze


KEY_DATA 
 DayNum 
 
 
 
 
 0 
 2021-12-03 
 4 
 
 
 1 
 2021-12-02 
 3 
 
 
 2 
 2021-12-01 
 2 
 
 
 3 
 2021-11-30 
 1 
 
 
 4 
 2021-11-29 
 0 
 
 
 ... 
 ... 
 ... 
 
 
 851 
 2019-08-04 
 6 
 
 
 852 
 2019-08-03 
 5 
 
 
 853 
 2019-08-02 
 4 
 
 
 854 
 2019-08-01 
 3 
 
 
 855 
 2019-07-31 
 2 
 
 
 
 856 rows × 2 columns

KEY_DATA DayNum 2021|12|03 4 2021|12|02 3 2021|12|01 2 2021|11|30 1 2021|11|29 0 2021|11|28 6 2021|11|27 5 2021|11|26 4 2021|11|25 3 2021|11|24 2 2021|11|23 1 2021|11|22 0 2021|11|21 6 2021|11|20 5 2021|11|19 4 2021|11|18 3 2021|11|17 2 2021|11|16 1 2021|11|15 0 2021|11|14 6 2021|11|13 5 2021|11|12 4 2021|11|11 3 2021|11|10 2 2021|11|09 1 2021|11|08 0 2021|11|07 6 2021|11|06 5 2021|11|05 4 2021|11|04 3 2021|11|03 2 2021|11|02 1 2021|11|01 0 2021|10|31 6 2021|10|30 5 2021|10|29 4 2021|10|28 3 2021|10|27 2 2021|10|26 1 2021|10|25 0 2021|10|24 6 2021|10|23 5 2021|10|22 4 2021|10|21 3 2021|10|20 2 2021|10|19 1 2021|10|18 0 2021|10|17 6 2021|10|16 5 2021|10|15 4 2021|10|14 3 2021|10|13 2 2021|10|12 1 2021|10|11 0 2021|10|10 6 2021|10|09 5 2021|10|08 4 2021|10|07 3 2021|10|06 2 2021|10|05 1 2021|10|04 0 2021|10|03 6 2021|10|02 5 2021|10|01 4 2021|09|30 3 2021|09|29 2 2021|09|28 1 2021|09|27 0 2021|09|26 6 2021|09|25 5 2021|09|24 4 2021|09|23 3 2021|09|22 2 2021|09|21 1 2021|09|20 0 2021|09|19 6 2021|09|18 5 2021|09|17 4 2021|09|16 3 2021|09|15 2 2021|09|14 1 2021|09|13 0 2021|09|12 6 2021|09|11 5 2021|09|10 4 2021|09|09 3 2021|09|08 2 2021|09|07 1 2021|09|06 0 2021|09|05 6 2021|09|04 5 2021|09|03 4 2021|09|02 3 2021|09|01 2 2021|08|31 1 2021|08|30 0 2021|08|29 6 2021|08|28 5 2021|08|27 4 2021|08|26 3 2021|08|25 2 2021|08|24 1 2021|08|23 0 2021|08|22 6 2021|08|21 5 2021|08|20 4 2021|08|19 3 2021|08|18 2 2021|08|17 1 2021|08|16 0 2021|08|15 6 2021|08|14 5 2021|08|13 4 2021|08|12 3 2021|08|11 2 2021|08|10 1 2021|08|09 0 2021|08|08 6 2021|08|07 5 2021|08|06 4 2021|08|05 3 2021|08|04 2 2021|08|03 1 2021|08|02 0 2021|08|01 6 2021|07|31 5 2021|07|30 4 2021|07|29 3 2021|07|28 2 2021|07|27 1 2021|07|26 0 2021|07|25 6 2021|07|24 5 2021|07|23 4 2021|07|22 3 2021|07|21 2 2021|07|20 1 2021|07|19 0 2021|07|18 6 2021|07|17 5 2021|07|16 4 2021|07|15 3 2021|07|14 2 2021|07|13 1 2021|07|12 0 2021|07|11 6 2021|07|10 5 2021|07|09 4 2021|07|08 3 2021|07|07 2 2021|07|06 1 2021|07|05 0 2021|07|04 6 2021|07|03 5 2021|07|02 4 2021|07|01 3 2021|06|30 2 2021|06|29 1 2021|06|28 0 2021|06|27 6 2021|06|26 5 2021|06|25 4 2021|06|24 3 2021|06|23 2 2021|06|22 1 2021|06|21 0 2021|06|20 6 2021|06|19 5 2021|06|18 4 2021|06|17 3 2021|06|16 2 2021|06|15 1 2021|06|14 0 2021|06|13 6 2021|06|12 5 2021|06|11 4 2021|06|10 3 2021|06|09 2 2021|06|08 1 2021|06|07 0 2021|06|06 6 2021|06|05 5 2021|06|04 4 2021|06|03 3 2021|06|02 2 2021|06|01 1 2021|05|31 0 2021|05|30 6 2021|05|29 5 2021|05|28 4 2021|05|27 3 2021|05|26 2 2021|05|25 1 2021|05|24 0 2021|05|23 6 2021|05|22 5 2021|05|21 4 2021|05|20 3 2021|05|19 2 2021|05|18 1 2021|05|17 0 2021|05|16 6 2021|05|15 5 2021|05|14 4 2021|05|13 3 2021|05|12 2 2021|05|11 1 2021|05|10 0 2021|05|09 6 2021|05|08 5 2021|05|07 4 2021|05|06 3 2021|05|05 2 2021|05|04 1 2021|05|03 0 2021|05|02 6 2021|05|01 5 2021|04|30 4 2021|04|29 3 2021|04|28 2 2021|04|27 1 2021|04|26 0 2021|04|25 6 2021|04|24 5 2021|04|23 4 2021|04|22 3 2021|04|21 2 2021|04|20 1 2021|04|19 0 2021|04|18 6 2021|04|17 5 2021|04|16 4 2021|04|15 3 2021|04|14 2 2021|04|13 1 2021|04|12 0 2021|04|11 6 2021|04|10 5 2021|04|09 4 2021|04|08 3 2021|04|07 2 2021|04|06 1 2021|04|05 0 2021|04|04 6 2021|04|03 5 2021|04|02 4 2021|04|01 3 2021|03|31 2 2021|03|30 1 2021|03|29 0 2021|03|28 6 2021|03|27 5 2021|03|26 4 2021|03|25 3 2021|03|24 2 2021|03|23 1 2021|03|22 0 2021|03|21 6 2021|03|20 5 2021|03|19 4 2021|03|18 3 2021|03|17 2 2021|03|16 1 2021|03|15 0 2021|03|14 6 2021|03|13 5 2021|03|12 4 2021|03|11 3 2021|03|10 2 2021|03|09 1 2021|03|08 0 2021|03|07 6 2021|03|06 5 2021|03|05 4 2021|03|04 3 2021|03|03 2 2021|03|02 1 2021|03|01 0 2021|02|28 6 2021|02|27 5 2021|02|26 4 2021|02|25 3 2021|02|24 2 2021|02|23 1 2021|02|22 0 2021|02|21 6 2021|02|20 5 2021|02|19 4 2021|02|18 3 2021|02|17 2 2021|02|16 1 2021|02|15 0 2021|02|14 6 2021|02|13 5 2021|02|12 4 2021|02|11 3 2021|02|10 2 2021|02|09 1 2021|02|08 0 2021|02|07 6 2021|02|06 5 2021|02|05 4 2021|02|04 3 2021|02|03 2 2021|02|02 1 2021|02|01 0 2021|0

### DataFrame Contatti

In [0]:
# File location and type
file_location = "/FileStore/LUMSA/Contatti.csv"
file_type = "csv"

# CSV options
infer_schema = "false"
first_row_is_header = "true"
delimiter = ";"

# The applied options are for CSV files. For other file types, these will be ignored.
contatti = spark.read.format(file_type) \
  .option("inferSchema", infer_schema) \
  .option("header", first_row_is_header) \
  .option("sep", delimiter) \
  .load(file_location)

display(contatti)

temp_table_name = "Contatti_csv"

contatti.createOrReplaceTempView(temp_table_name)

# Numero di osservazioni
print(f'Il DF contatti contiene: {contatti.count()} occorrenze')

# Non sembrano esserci campi anomali tranne i valori 'null' spesso ripetuti che però Pandas gestisce come 'None'
CONTATTI = contatti.toPandas()
CONTATTI

Il DF contatti contiene: 65244 occorrenze


contactid 
 EducationCodeValue 
 GenderCodeValue 
 RegioneDiResidenza 
 StatoOccupazionale 
 
 
 
 
 0 
 00006912-4290-ea11-a811-000d3a23cc7b 
 Laurea 
 Femminile 
 Piemonte 
 Libero professionista 
 
 
 1 
 0000c765-d180-ea11-a811-000d3a21188e 
 Licenza media 
 Femminile 
 Puglia 
 Disoccupato 
 
 
 2 
 00012cd8-3ea7-eb11-b1ac-000d3ac251b6 
 None 
 None 
 None 
 None 
 
 
 3 
 00021b17-7438-eb11-a813-000d3a3a1391 
 None 
 Femminile 
 None 
 None 
 
 
 4 
 000285f4-9581-eb11-a812-000d3abd3d02 
 None 
 None 
 None 
 None 
 
 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 
 
 65239 
 fffa53fd-69e9-eb11-bacb-000d3a48e944 
 None 
 Maschile 
 None 
 None 
 
 
 65240 
 fffc2b1a-0741-ea11-a813-000d3a23c1dc 
 None 
 Maschile 
 Puglia 
 None 
 
 
 65241 
 fffd9c36-ba4a-eb11-a812-000d3a21188e 
 None 
 Maschile 
 None 
 None 
 
 
 65242 
 fffdccaf-d7c6-ea11-a812-000d3a21188e 
 None 
 Femminile 
 None 
 None 
 
 
 65243 
 ffffbfcd-992c-eb11-a813-000d3a23cc7b 
 None 
 Maschile 
 None 
 None 
 
 
 
 65244 rows × 5 columns

contactid EducationCodeValue GenderCodeValue RegioneDiResidenza StatoOccupazionale 00006912-4290-ea11-a811-000d3a23cc7b Laurea Femminile Piemonte Libero professionista 0000c765-d180-ea11-a811-000d3a21188e Licenza media Femminile Puglia Disoccupato 00012cd8-3ea7-eb11-b1ac-000d3ac251b6 null null null null 00021b17-7438-eb11-a813-000d3a3a1391 null Femminile null null 000285f4-9581-eb11-a812-000d3abd3d02 null null null null 00059e95-8ef6-eb11-94ef-00224884aa9c Licenza superiore Maschile Lazio Tempo indeterminato 00074734-abdc-4445-85ed-d6ae3e0d9ef7 Master L1 Femminile Campania Tempo indeterminato 00078c34-3886-ea11-a811-000d3a21188e null Maschile null null 00078ce5-f23f-ec11-8c62-6045bd8d9ef6 Laurea Maschile Lazio Tempo indeterminato 00093f79-1169-ea11-a811-000d3a23cc7b Licenza superiore Maschile Campania Imprenditore 0009cbf5-4823-eb11-a813-000d3a23cc7b null Femminile null null 0009e973-7920-eb11-a813-000d3a21188e null Femminile Sardegna null 000c8180-8240-eb11-a813-000d3a21188e Laurea Maschile Sicilia Libero professionista 000c990c-b757-ea11-a811-000d3a24f1d2 Laurea Maschile Lombardia Dirigente 000d020f-f2f5-ea11-a815-000d3a23cc7b Licenza media Femminile Sicilia Disoccupato 000dc36f-e04f-eb11-a812-000d3a21188e null Maschile Campania null 000dd486-476c-eb11-a812-000d3a391feb null null null null 000df6a2-6e9c-eb11-b1ac-000d3a2d98e3 Laurea Maschile Campania Dirigente 000ec335-952a-eb11-a813-000d3a23cc7b Licenza media Femminile Toscana Imprenditore 00117bf2-ef19-ec11-b6e6-6045bd8de0b6 Master L1 Maschile Piemonte Tempo indeterminato 0011a75c-f32b-4323-8290-2990ef78682b Laurea Femminile Marche Tempo indeterminato 00120898-4c8c-ea11-a811-000d3a21188e null Maschile null null 001559dd-c669-ea11-a811-000d3a23cc7b null Femminile null null 00156c8d-ac78-e911-a971-000d3a38c9b5 null Maschile Campania null 0015f547-872d-eb11-a813-000d3a23cc7b Laurea Maschile Campania Imprenditore 00164e47-df95-ea11-a811-000d3a21188e null Maschile null null 0018d788-bba2-eb11-b1ac-000d3a2ba9ce Licenza media Femminile null Disoccupato 001908b7-961a-ea11-a811-000d3a23c07f null Femminile null null 0019b469-6402-eb11-a813-000d3a21188e null Maschile null null 001aea7c-15a4-eb11-b1ac-000d3a2ba9ce Licenza superiore Maschile Sicilia Tempo indeterminato 001b07f0-07f5-eb11-94ef-000d3a2bf457 null null null null 001e5c72-0afb-e911-a813-000d3a23cb53 null null null null 001e9468-60e4-e911-a813-000d3a24ed81 null null null null 002055f9-0bb2-e911-a975-000d3a38c9b5 null Maschile null null 00208ee1-b26c-eb11-a812-000d3a391feb Licenza superiore Maschile null Libero professionista 0021ea09-5d19-eb11-a813-000d3a21188e null null null null 0021f6af-a98a-eb11-b1ac-000d3aac7941 Laurea Maschile Puglia Imprenditore 00227c34-7e8f-ea11-a811-000d3a21188e Laurea Femminile Emilia-Romagna Tempo indeterminato 00231c5f-2b23-eb11-a813-000d3a23cc7b null Femminile null null 0024a68e-7d82-eb11-a812-000d3abd3d02 Licenza superiore Femminile Marche Studente 00265b4c-dff8-ea11-a815-000d3a21188e Licenza superiore Maschile Campania Disoccupato 002760af-af78-e911-a97e-000d3a38c9f2 null Maschile Calabria null 00284a45-14c7-4391-b702-325f888feb36 null Maschile Sicilia null 0028c661-6663-ea11-a811-000d3a24f1d2 Licenza superiore Maschile Sardegna Altro 00295b0a-de2b-eb11-a813-000d3a23cc7b null Femminile null null 002a8869-dadb-e911-a812-000d3a24f3f0 Laurea Maschile Campania Consulente 002c9a5e-3df6-e911-a813-000d3a24e147 null null null null 002ceb53-059f-eb11-b1ac-000d3a2ba9ce null null null null 002d3e91-943b-ea11-a813-000d3a23cde2 Laurea Maschile Veneto Imprenditore 002e0c1c-0b96-ea11-a811-000d3a23cc7b Laurea Maschile Lazio Libero professionista 002ed87b-b00e-eb11-a813-000d3a21188e Laurea Maschile Lombardia Dirigente 002ef8f6-184c-ea11-a813-000d3a23cde2 Laurea Maschile Lazio Tempo indeterminato 002f3bd6-1efc-ea11-a813-000d3a23cc7b null null Campania null 0030565f-fec8-e911-a976-000d3a38c9b5 null Femminile null null 00315ae1-b280-eb11-a812-000d3abd3d02 Licenza media Maschile Sicilia Disoccupato 00316431-4736-

### DataFrame DecodificaOrigineCaso

In [0]:
# File location and type
file_location = "/FileStore/LUMSA/DecodificaOrigineCaso.csv"
file_type = "csv"

# CSV options
infer_schema = "false"
first_row_is_header = "true"
delimiter = ";"

# The applied options are for CSV files. For other file types, these will be ignored.
decodificaOrigineCaso = spark.read.format(file_type) \
  .option("inferSchema", infer_schema) \
  .option("header", first_row_is_header) \
  .option("sep", delimiter) \
  .load(file_location)

display(decodificaOrigineCaso)

temp_table_name = "DecodificaOrigineCaso_csv"

decodificaOrigineCaso.createOrReplaceTempView(temp_table_name)

# Numero di osservazioni
print(f'Il DF decodificaOrigineCaso contiene: {decodificaOrigineCaso.count()} occorrenze')

# Portiamo l'ultimo dataset in formato Pandas per poterli concatenare tutti con i join
DECODIFICA_ORIGINE_CASO = decodificaOrigineCaso.toPandas()
DECODIFICA_ORIGINE_CASO

Il DF decodificaOrigineCaso contiene: 2 occorrenze


ID 
 Valore 
 
 
 
 
 0 
 3 
 Schede Contatto 
 
 
 1 
 1 
 Telefono

ID Valore 3 Schede Contatto 1 Telefono

### 6. Riassunto delle numerosità di ogni DataFrame

In [0]:
# Riassunto delle numerosità di ogni dataset
print(f'Il DF casi contiene: {casi.count()} occorrenze')
print(f'Il DF calendario contiene: {calendario.count()} occorrenze')
print(f'Il DF contatti contiene: {contatti.count()} occorrenze')
print(f'Il DF decodificaOrigineCaso contiene: {decodificaOrigineCaso.count()} occorrenze')

Il DF casi contiene: 149997 occorrenze
Il DF calendario contiene: 856 occorrenze
Il DF contatti contiene: 65244 occorrenze
Il DF decodificaOrigineCaso contiene: 2 occorrenze


### 7a. Ridenominazione dei giorni della settimana

In [0]:
Calendario_DAY['day_of_week'] = Calendario_DAY['KEY_DATA'].dt.day_name()
Calendario_DAY

KEY_DATA 
 DayNum 
 day_of_week 
 
 
 
 
 0 
 2021-12-03 
 4 
 Friday 
 
 
 1 
 2021-12-02 
 3 
 Thursday 
 
 
 2 
 2021-12-01 
 2 
 Wednesday 
 
 
 3 
 2021-11-30 
 1 
 Tuesday 
 
 
 4 
 2021-11-29 
 0 
 Monday 
 
 
 ... 
 ... 
 ... 
 ... 
 
 
 851 
 2019-08-04 
 6 
 Sunday 
 
 
 852 
 2019-08-03 
 5 
 Saturday 
 
 
 853 
 2019-08-02 
 4 
 Friday 
 
 
 854 
 2019-08-01 
 3 
 Thursday 
 
 
 855 
 2019-07-31 
 2 
 Wednesday 
 
 
 
 856 rows × 3 columns

### 7b. Numero totale di Ticket

In [0]:
print(f'Il DF casi contiene: {casi.count()} occorrenze')

Il DF casi contiene: 149997 occorrenze


In [0]:
%sql

select COUNT(*) from `Casi_csv`

count(1) 149997

In [0]:
# Per controllare gli attributi di ogni DF
print(CASI.columns)
print(CONTATTI.columns)
print(Calendario_DAY.columns)
print(DECODIFICA_ORIGINE_CASO.columns)

Index(['customerid', 'LivelloEscalation', 'KEY_DATA', 'OraArrivo',
       'NumeroTicket', 'DataArrivo', 'OrigineCaso', 'ClassificazioneTicket'],
      dtype='object')
Index(['contactid', 'EducationCodeValue', 'GenderCodeValue',
       'RegioneDiResidenza', 'StatoOccupazionale'],
      dtype='object')
Index(['KEY_DATA', 'DayNum', 'day_of_week'], dtype='object')
Index(['ID', 'Valore'], dtype='object')


### 8. Unione delle tabelle in una temporary table ("Casi")

In [0]:
# Left-outer join su KEY_DATA poiché la table principale è quella dei casi
newCasi = pd.merge(CASI,Calendario_DAY,on='KEY_DATA',how='left')
# Per eseguire un left-outer join su 'customerid', provvediamo a ridenominare la colonna 'contactid' di CONTATTI per poi eseguire un merge
CONTATTI.rename(columns={'contactid': 'customerid'}, inplace=True)
# Eseguiamo il merge sull'attributo comune 'customerid'
newCasi = pd.merge(newCasi,CONTATTI,on='customerid',how='left')
# Per quanto riguarda la DecodificaOrigineCaso, provvediamo sempre ad eseguire prima una ridenominazione e successivamente il merge
DECODIFICA_ORIGINE_CASO.rename(columns={'ID': 'OrigineCaso'}, inplace=True)
newCasi = pd.merge(newCasi,DECODIFICA_ORIGINE_CASO,on='OrigineCaso',how='left')
# Ci troviamo davanti al caso in cui abbiamo ridondanza di informazioni (attributi duplicati). Confrontiamo KEY_DATA con DataArrivo per verificare che siano uguali, in tal caso procederemo ad effettuare un drop di una delle due
if len(newCasi[newCasi['KEY_DATA'] == newCasi['DataArrivo']]) == len(newCasi):
  print("Drop needed")
newCasi = newCasi.drop(columns=['KEY_DATA'])
# Anche il giorno della settimana è duplicato, teniamo il valore categorico piuttosto che il discreto per rappresentazioni quali hist et simil.
newCasi = newCasi.drop(columns=['DayNum'])
# Stesso discorso per OrigineCaso che può essere dedotto da Valore che rinomineremo a 'OrigineCaso'
newCasi = newCasi.drop(columns=['OrigineCaso'])
newCasi.rename(columns={'Valore': 'OrigineCaso'}, inplace=True)
# Riordiniamo le colonne in base al tipo di dato
newCasi = newCasi[['customerid', 'NumeroTicket', 'ClassificazioneTicket', 'OrigineCaso', 'day_of_week', 'DataArrivo', 'OraArrivo', 'EducationCodeValue','GenderCodeValue','RegioneDiResidenza','StatoOccupazionale']]
# Abbiamo deciso di rinominare gli attributi rendendoli 'capitalized' -->
newCasi.rename(columns={'customerid': 'CustomerID'}, inplace=True)
newCasi.rename(columns={'day_of_week': 'DayOfWeek'}, inplace=True)
# Onde evitare problemi con il riconoscimento dell'ora come stringa, trattiamo l'ora come un intero
newCasi['OraArrivo'] = pd.to_numeric(newCasi['OraArrivo'])
newCasi

Drop needed


CustomerID 
 NumeroTicket 
 ClassificazioneTicket 
 OrigineCaso 
 DayOfWeek 
 DataArrivo 
 OraArrivo 
 EducationCodeValue 
 GenderCodeValue 
 RegioneDiResidenza 
 StatoOccupazionale 
 
 
 
 
 0 
 00006912-4290-ea11-a811-000d3a23cc7b 
 TCK-52759-W1C0 
 Informazioni 
 Schede Contatto 
 Thursday 
 2020-05-07 
 14 
 Laurea 
 Femminile 
 Piemonte 
 Libero professionista 
 
 
 1 
 0000c765-d180-ea11-a811-000d3a21188e 
 TCK-45038-K4C6 
 Informazioni 
 Schede Contatto 
 Monday 
 2020-04-20 
 10 
 Licenza media 
 Femminile 
 Puglia 
 Disoccupato 
 
 
 2 
 00012cd8-3ea7-eb11-b1ac-000d3ac251b6 
 TCK-192329-L0X9 
 Informazioni 
 Schede Contatto 
 Tuesday 
 2021-04-27 
 14 
 None 
 None 
 None 
 None 
 
 
 3 
 00021b17-7438-eb11-a813-000d3a3a1391 
 TCK-130028-X9N6 
 Informazioni 
 Telefono 
 Monday 
 2020-12-07 
 11 
 None 
 Femminile 
 None 
 None 
 
 
 4 
 000285f4-9581-eb11-a812-000d3abd3d02 
 TCK-169373-C9N9 
 Informazioni 
 Schede Contatto 
 Thursday 
 2021-03-11 
 11 
 None 
 None 
 None 
 None 
 
 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 ... 
 
 
 149992 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 TCK-162029-G8M5 
 Informazioni 
 Schede Contatto 
 Wednesday 
 2021-02-24 
 13 
 None 
 None 
 None 
 None 
 
 
 149993 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 TCK-172017-X1L7 
 Informazioni 
 Schede Contatto 
 Wednesday 
 2021-03-17 
 12 
 None 
 None 
 None 
 None 
 
 
 149994 
 8ff902bd-df55-eb11-a812-000d3a21188e 
 TCK-190698-G3J7 
 Informazioni 
 Schede Contatto 
 Friday 
 2021-04-23 
 16 
 None 
 None 
 None 
 None 
 
 
 149995 
 8ff976a2-4be4-e911-a813-000d3a24e3ab 
 TCK-61020-V9N8 
 Informazioni 
 Telefono 
 Monday 
 2020-05-25 
 13 
 Licenza superiore 
 Maschile 
 Campania 
 Imprenditore 
 
 
 149996 
 8ff976a2-4be4-e911-a813-000d3a24e3ab 
 TCK-65209-R6X4 
 Informazioni 
 Telefono 
 Tuesday 
 2020-06-09 
 17 
 Licenza superiore 
 Maschile 
 Campania 
 Imprenditore 
 
 
 
 149997 rows × 11 columns

In [0]:
# Prima riconvertiamo la struttura dati da dataframe Pandas a dataframe Spark
Casi = spark.createDataFrame(newCasi)
# Generiamo una temporary table per elaborare i dati
temp_table_name = "Casi_pandas"
Casi.createOrReplaceTempView(temp_table_name)

### 9. BarChart ticket in funzione del giorno della settimana

In [0]:
%sql
select COUNT(NumeroTicket) as `total_ticket`, `DayOfWeek` as `Day` 
from `Casi_pandas` 
group by `DayOfWeek` 
order by `total_ticket` DESC

total_ticket Day 31553 Tuesday 31195 Monday 29836 Wednesday 29005 Thursday 25180 Friday 1950 Saturday 1278 Sunday

## Risposte alle domande:
#### a) Qual è il giorno con il maggior numero di ticket? 
Tuesday (Martedì)
#### b) Qual è il giorno con il minor numero di ticket? 
Sunday (Domenica)
#### c) Quantifica il numero di ticket per i 2 giorni sopra indicati. 
Martedì (31.553k); Domenica (1278)

### -INFO- 
#### Approfondendo sull'origine dei casi, ci è sembrato opportuno controllare quanti di questi provenissero da Schede Contatto o Telefono. Il risultato è che il rapporto "Telefono:Scheda Contatto" è circa 2:1 in tutti i giorni lavorativi. Durante il weekend, invece, le schede contatto sono la principale origine dei casi.

In [0]:
%sql
select COUNT(NumeroTicket) as `total_ticket`, `DayOfWeek` as `Day`, `OrigineCaso`
from `Casi_pandas` 
group by `DayOfWeek`,`OrigineCaso`
order by `total_ticket` DESC

total_ticket Day OrigineCaso 20726 Monday Telefono 20363 Tuesday Telefono 19470 Wednesday Telefono 19045 Thursday Telefono 16518 Friday Telefono 11190 Tuesday Schede Contatto 10469 Monday Schede Contatto 10366 Wednesday Schede Contatto 9960 Thursday Schede Contatto 8662 Friday Schede Contatto 1834 Saturday Schede Contatto 1234 Sunday Schede Contatto 116 Saturday Telefono 44 Sunday Telefono

### -INFO- 
#### Sulla tipologia di Ticket c'è da dire che la maggior parte sono di "informazione", seguiti da "reclamo" e "caduta linea" durante tutti i giorni della settimana.

In [0]:
%sql
select COUNT(NumeroTicket) as `total_ticket`, `DayOfWeek` as `Day`, `ClassificazioneTicket`
from `Casi_pandas` 
group by `DayOfWeek`,`ClassificazioneTicket`
order by `total_ticket` DESC

total_ticket Day ClassificazioneTicket 27718 Monday Informazioni 27512 Tuesday Informazioni 26588 Wednesday Informazioni 26138 Thursday Informazioni 22867 Friday Informazioni 2187 Tuesday Reclamo 1886 Saturday Informazioni 1636 Wednesday Reclamo 1607 Monday Reclamo 1325 Thursday Reclamo 1243 Sunday Informazioni 1013 Friday Reclamo 940 Monday Caduta Linea 854 Tuesday Caduta Linea 784 Thursday Caduta Linea 780 Wednesday Caduta Linea 607 Friday Caduta Linea 479 Tuesday Presa Veloce 401 Monday Orientamento 398 Monday Presa Veloce 390 Wednesday Orientamento 388 Tuesday Orientamento 387 Thursday Orientamento 351 Friday Orientamento 309 Wednesday Presa Veloce 244 Thursday Presa Veloce 244 Friday Presa Veloce 126 Monday N.D. 121 Thursday N.D. 120 Wednesday N.D. 117 Tuesday N.D. 94 Friday N.D. 53 Saturday Reclamo 31 Sunday Reclamo 16 Tuesday Callback 13 Wednesday Callback 9 Saturday Orientamento 6 Thursday Callback 5 Monday Callback 4 Sunday Orientamento 4 Friday Callback 1 Saturday Caduta Linea 1 Saturday N.D.

### 10. TrendLine del quantità di ticket (divisi per tipo) per fascia oraria

In [0]:
%sql
select COUNT(NumeroTicket) as `total_ticket`, `OraArrivo` as `Hour`, `ClassificazioneTicket`
from `Casi_pandas` 
group by `OraArrivo`, `ClassificazioneTicket`
order by `total_ticket` DESC

total_ticket Hour ClassificazioneTicket 19496 11 Informazioni 18445 10 Informazioni 16792 12 Informazioni 14267 16 Informazioni 13478 9 Informazioni 13152 17 Informazioni 12533 15 Informazioni 8328 14 Informazioni 7719 13 Informazioni 3992 18 Informazioni 1660 19 Informazioni 1322 10 Reclamo 1240 11 Reclamo 1129 8 Informazioni 1100 12 Reclamo 859 9 Reclamo 741 15 Reclamo 711 20 Informazioni 678 10 Caduta Linea 667 16 Reclamo 667 11 Caduta Linea 597 9 Caduta Linea 539 13 Reclamo 536 14 Reclamo 533 21 Informazioni 517 17 Reclamo 468 12 Caduta Linea 460 22 Informazioni 459 16 Caduta Linea 375 23 Informazioni 350 10 Presa Veloce 329 11 Orientamento 312 17 Caduta Linea 292 15 Caduta Linea 286 11 Presa Veloce 282 7 Informazioni 273 10 Orientamento 262 12 Orientamento 249 14 Caduta Linea 240 12 Presa Veloce 227 16 Presa Veloce 224 13 Caduta Linea 209 0 Informazioni 197 9 Presa Veloce 191 15 Orientamento 188 17 Orientamento 181 9 Orientamento 177 16 Orientamento 160 18 Reclamo 141 13 Orientamento 140 14 Orientamento 119 1 Informazioni 110 14 Presa Veloce 109 11 N.D. 105 15 Presa Veloce 95 17 Presa Veloce 90 6 Informazioni 89 10 N.D. 81 12 N.D. 77 2 Informazioni 65 16 N.D. 61 13 Presa Veloce 58 15 N.D. 58 19 Reclamo 49 8 Reclamo 46 5 Informazioni 45 9 N.D. 44 17 N.D. 37 14 N.D. 33 13 N.D. 31 4 Informazioni 28 3 Informazioni 20 18 Caduta Linea 19 20 Reclamo 15 18 Orientamento 14 13 Callback 11 22 Reclamo 11 7 Reclamo 10 14 Callback 8 21 Reclamo 8 18 N.D. 6 0 Reclamo 6 11 Callback 5 21 Orientamento 5 12 Callback 4 23 Reclamo 4 19 Orientamento 4 23 Orientamento 4 22 Orientamento 4 0 Orientamento 3 1 Orientamento 3 20 Orientamento 3 8 Orientamento 3 1 Reclamo 3 18 Presa Veloce 2 21 N.D. 2 6 Reclamo 2 8 N.D. 2 15 Callback 2 9 Callback 2 16 Callback 2 19 N.D. 2 17 Callback 2 7 Orientamento 1 0 N.D. 1 4 Orientamento 1 20 N.D. 1 23 N.D. 1 22 N.D. 1 10 Callback

### Considerazioni sul grafico precedente (10)
Tutti gli andamenti mostrano un trend crescente tra le 6 e le 11, un flesso tra le 11 e le 14, un nuovo trend crescente tra le 14 e le 16 per poi terminare con un calo fino alle 22. Già dall'istogramma fatto in precedenza era visibile che i ticket per informazioni sono in quantità nettamente superiore al resto dei ticket ma rappresentando l'andamento nelle ore del giorno si nota come vi siano due picchi in corrispondenza delle ore 11 e delle ore 16. Anche le curve con cardinalità di ticket minore seguono lo stesso andamento anche se in maniera meno pedissequa. C'è sicuramente un effetto interessante da analizzare tra le ore 13 e 14 probabilmente dovuto a qualche fattore esterno non in nostro possesso.

### 11a. Top 5 regioni per numero di Ticket

In [0]:
%sql
CREATE OR REPLACE TABLE topRegion
select COUNT(NumeroTicket) as `total_ticket`, `RegioneDiResidenza` as `Regione`
from `Casi_pandas`
where `OrigineCaso`="Telefono" AND `RegioneDiResidenza` IS NOT NULL
group by Regione
order by `total_ticket` DESC
LIMIT 5

num_affected_rows num_inserted_rows

### 11b. Percentuale di Ticket per Top 5 regioni e restanti

In [0]:
%sql
select SUM(total_ticket) as total_ticket
from (select COUNT(NumeroTicket) as `total_ticket`, `RegioneDiResidenza` as `Regione`
from `Casi_pandas`
where `OrigineCaso`="Telefono" AND `RegioneDiResidenza` IS NOT NULL
group by Regione
order by `total_ticket` ASC
LIMIT 15)

total_ticket 13359

In [0]:
%sql
INSERT INTO topRegion 
VALUES (13359, 'Altre')

num_affected_rows num_inserted_rows 1 1

In [0]:
%sql 
SELECT * FROM topRegion

total_ticket Regione 5608 Campania 5323 Sicilia 3621 Lazio 2990 Lombardia 2336 Puglia 13359 Altre

### 12. Quanti sono i giorni di attività del Contact Center durante tutto il periodo considerato? (attraverso il dataset Casi)

In [0]:
print(f"I giorni di attività del Contact Center durante tutto il periodo considerato sono {len(newCasi['DataArrivo'].unique().tolist())}.")

### 13. Quali sono i 3 giorni (es. lunedì xx/11/2020) con il maggior numero assoluto di casi, cioè i giorni in cui si verificano dei picchi di lavorazione?
I giorni con il maggior numero assoluto di casi sono (in ordine decrescente):
  * il 28/09/2021 per un totale di 1188 ticket aperti
  * il 23/11/2020 per un totale di 748 ticket aperti
  * il 20/11/2020 per un totale di 654 ticket aperti

In [0]:
%sql

select COUNT(NumeroTicket) as `total_ticket`, `DataArrivo`
from `Casi_pandas`
group by `DataArrivo`
order by `total_ticket` DESC
LIMIT 3

### 14. Quale fascia oraria mediamente riceve più telefonate?
La fascia oraria che riceve più telefonate è quella dalle 9:00 alle 12:00 A.M.

### 15. Qual è la percentuale di casi per ogni regione?

In [0]:
diz = dict(round(newCasi['RegioneDiResidenza'].value_counts(normalize=True)*100,2))
for key in diz.keys():
  print(f'La percentuale dei Ticket della regione {key} è {diz.get(key)}%')
  


La percentuale dei Ticket della regione Campania è 16.98%
La percentuale dei Ticket della regione Sicilia è 12.77%
La percentuale dei Ticket della regione Lombardia è 10.8%
La percentuale dei Ticket della regione Lazio è 9.42%
La percentuale dei Ticket della regione Puglia è 7.27%
La percentuale dei Ticket della regione Veneto è 5.6%
La percentuale dei Ticket della regione Calabria è 5.59%
La percentuale dei Ticket della regione Emilia-Romagna è 5.39%
La percentuale dei Ticket della regione Toscana è 4.43%
La percentuale dei Ticket della regione Piemonte è 4.43%
La percentuale dei Ticket della regione Sardegna è 3.99%
La percentuale dei Ticket della regione Abruzzo è 3.57%
La percentuale dei Ticket della regione Marche è 2.91%
La percentuale dei Ticket della regione Basilicata è 1.51%
La percentuale dei Ticket della regione Friuli-Venezia Giulia è 1.32%
La percentuale dei Ticket della regione Liguria è 1.29%
La percentuale dei Ticket della regione Umbria è 1.23%
La percentuale dei Tick

### 16. Qual è la percentuale per genere, livello di istruzione e stato occupazionale declinata per telefonate e schede contatto? (tipologia di origine caso -> è possibile filtrare via i valori nulli per ogni per genere quando la percentuale da calcolare è genere, e così via)

In [0]:
%sql
select COUNT(NumeroTicket) as total_ticket, GenderCodeValue, OrigineCaso
from `Casi_pandas`
where `GenderCodeValue` IS NOT NULL
group by `GenderCodeValue`, OrigineCaso

total_ticket GenderCodeValue OrigineCaso 15439 Femminile Schede Contatto 28412 Maschile Schede Contatto 16843 Femminile Telefono 35916 Maschile Telefono

In [0]:
%sql
select COUNT(NumeroTicket) as total_ticket, EducationCodeValue, OrigineCaso
from `Casi_pandas`
where `EducationCodeValue` IS NOT NULL
group by `EducationCodeValue`, OrigineCaso

total_ticket EducationCodeValue OrigineCaso 1495 Dottorato Telefono 99 Nessuno Telefono 14015 Laurea Schede Contatto 132 Nessuno Schede Contatto 10325 Laurea Telefono 741 Dottorato Schede Contatto 100 Licenza elementare Schede Contatto 1335 Licenza media Telefono 2285 Licenza media Schede Contatto 1319 Master L1 Schede Contatto 1242 Master L2 Schede Contatto 1007 Master L1 Telefono 7471 Licenza superiore Telefono 796 Master L2 Telefono 49 Licenza elementare Telefono 11624 Licenza superiore Schede Contatto

In [0]:
%sql
select COUNT(NumeroTicket) as total_ticket, StatoOccupazionale, OrigineCaso
from `Casi_pandas`
where `StatoOccupazionale` IS NOT NULL
group by `StatoOccupazionale`, OrigineCaso

total_ticket StatoOccupazionale OrigineCaso 5160 Libero professionista Schede Contatto 4966 Disoccupato Schede Contatto 1182 Altro Telefono 43 Lavoratore interinale Schede Contatto 110 Ricercatore Schede Contatto 535 Tempo determinato Telefono 7841 Imprenditore Telefono 387 Studente Telefono 5030 Tempo indeterminato Schede Contatto 15 Lavoratore interinale Telefono 341 Dirigente Telefono 2574 Consulente Schede Contatto 3236 Disoccupato Telefono 192 Inoccupato (neet) Telefono 1657 Consulente Telefono 744 Dirigente Schede Contatto 70 Ricercatore Telefono 324 Inoccupato (neet) Schede Contatto 8558 Imprenditore Schede Contatto 581 Studente Schede Contatto 2198 Altro Schede Contatto 4320 Libero professionista Telefono 2114 Tempo indeterminato Telefono 1163 Tempo determinato Schede Contatto